### PR TITLE
pan-os edl migration guide

### PR DESCRIPTION
**Pull request by bakatzir**
## Status
- [x] In Progress

## Related Issues
fixes: https://github.com/demisto/etc/issues/35212

## Minimum version of Demisto
- [x] 5.0.0

## Does it break backward compatibility?
   - [x] No

## Must have
- [x] Documentation

### CODE REVIEW

**Comment by bakatzir** [comment]
@ytauman @DeanArbel please follow the migration steps I've added and comment inline so Ican tend to whatever is needed.\
@kirbles19 I'll ask you to review it afterwards as we get multiple requests about this.

File: Packs/PaloAltoNetworks_PAN_OS_EDL_Management/Integrations/PaloAltoNetworks_PAN_OS_EDL_Management/PaloAltoNetworks_PAN_OS_EDL_Management_description.md [hunk 1]
```diff
@@ -1,10 +1,12 @@
...
-   - **CentOS**: `"httpd -S"` 
-   - **Ubuntu**: `apcahe2 -S"`
+This integration is deprecated, please refer to [PAN-OS EDL](https://xsoar.pan.dev/docs/reference/integrations/edl) to learn more about hosting EDLs in Cortex XSOAR. 
+
+## Set Up a Remote Web Server
+To use the Palo Alto Networks PAN-OS EDL Management integration, you need to set up a remote web server.
+1. Set up a remote server with Apache.
+2. Generate a pair of SSH keys. Send the private key into the user’s home directory, into “.ssh” folder in the Apache server. 
+Append it to the “authorized_keys” file.
+3. Save the private SSH key in Demisto Credentials.
```
Comment (ytauman) [inline Packs/PaloAltoNetworks_PAN_OS_EDL_Management/Integrations/PaloAltoNetworks_PAN_OS_EDL_Management/PaloAltoNetworks_PAN_OS_EDL_Management_description.md:8]: isn't this supposed to be the public key?
plus - remove the word 'SSH' (it's not mentioned in the previous step either).

File: Packs/PaloAltoNetworks_PAN_OS_EDL_Management/Integrations/PaloAltoNetworks_PAN_OS_EDL_Management/PaloAltoNetworks_PAN_OS_EDL_Management_description.md [hunk 2]
```diff
@@ -1,10 +1,12 @@
...
- 3. Save the private SSH key in Demisto Credentials.
- 4. To verify the location of the document root where the files are stored, run the following command.
-   - **CentOS**: `"httpd -S"` 
-   - **Ubuntu**: `apcahe2 -S"`
+This integration is deprecated, please refer to [PAN-OS EDL](https://xsoar.pan.dev/docs/reference/integrations/edl) to learn more about hosting EDLs in Cortex XSOAR. 
+
+## Set Up a Remote Web Server
+To use the Palo Alto Networks PAN-OS EDL Management integration, you need to set up a remote web server.
+1. Set up a remote server with Apache.
+2. Generate a pair of SSH keys. Send the private key into the user’s home directory, into “.ssh” folder in the Apache server.
```
Comment (ytauman) [inline Packs/PaloAltoNetworks_PAN_OS_EDL_Management/Integrations/PaloAltoNetworks_PAN_OS_EDL_Management/PaloAltoNetworks_PAN_OS_EDL_Management_description.md:6]: Send >> Save ?

File: Packs/PaloAltoNetworks_PAN_OS_EDL_Management/Integrations/PaloAltoNetworks_PAN_OS_EDL_Management/PaloAltoNetworks_PAN_OS_EDL_Management_description.md [hunk 3]
```diff
@@ -1,10 +1,12 @@
...
-   - **Ubuntu**: `apcahe2 -S"`
+This integration is deprecated, please refer to [PAN-OS EDL](https://xsoar.pan.dev/docs/reference/integrations/edl) to learn more about hosting EDLs in Cortex XSOAR. 
+
+## Set Up a Remote Web Server
+To use the Palo Alto Networks PAN-OS EDL Management integration, you need to set up a remote web server.
+1. Set up a remote server with Apache.
+2. Generate a pair of SSH keys. Send the private key into the user’s home directory, into “.ssh” folder in the Apache server. 
+Append it to the “authorized_keys” file.
+3. Save the private SSH key in Demisto Credentials.
+4. To verify the location of the document root where the files are stored, run the following command.
```
Comment (ytauman) [inline Packs/PaloAltoNetworks_PAN_OS_EDL_Management/Integrations/PaloAltoNetworks_PAN_OS_EDL_Management/PaloAltoNetworks_PAN_OS_EDL_Management_description.md:9]: not sure what 'document root' means in this context..?

File: Packs/EDL/Integrations/EDL/README.md [hunk 4]
```diff
@@ -1,13 +1,22 @@
 This integration provides External Dynamic List (EDL) as a service for the system indicators (Outbound feed). The feed content can then be sent to 3rd parties for improving their security alignment.
 
+## PAN-OS EDL Management to PAN-OS EDL Service migration steps
+This integration hosts the EDL on the Cortex XSOAR server. Please follow these steps to migrate your EDLs.
```
Comment (ytauman) [inline Packs/EDL/Integrations/EDL/README.md:4]: I would replace the beginning of the sentence with `Unlike EDL Management, EDL Service hosts the EDL on...`
This way we explain one of the main differences (hosting the list locally and not with a remote server).

File: Packs/EDL/Integrations/EDL/README.md [hunk 5]
```diff
@@ -1,13 +1,22 @@
...
+1. Create IOCs in Cortex XSOAR from the EDLs data, which was previously hosted on an external web server. To do so automatically, extract your EDL as a text file from the web server, upload it as a file to the war-room and use the `ExtractIndicatorsFromTextFile` automation. e.g, `!ExtractIndicatorsFromTextFile entryID=<entry_id>`
```
Comment (ytauman) [inline Packs/EDL/Integrations/EDL/README.md:5]: Would rephrase to:
`Convert existing EDL list to indicators in Cortex XSOAR. This can be done automatically: extract your EDL as a text file from the web server it's currently hosted on, upload it as a file to the Playground, and use the...`

File: Packs/EDL/Integrations/EDL/README.md [hunk 6]
```diff
@@ -1,13 +1,22 @@
 This integration provides External Dynamic List (EDL) as a service for the system indicators (Outbound feed). The feed content can then be sent to 3rd parties for improving their security alignment.
 
+## PAN-OS EDL Management to PAN-OS EDL Service migration steps
+This integration hosts the EDL on the Cortex XSOAR server. Please follow these steps to migrate your EDLs.
+1. Create IOCs in Cortex XSOAR from the EDLs data, which was previously hosted on an external web server. To do so automatically, extract your EDL as a text file from the web server, upload it as a file to the war-room and use the `ExtractIndicatorsFromTextFile` automation. e.g, `!ExtractIndicatorsFromTextFile entryID=<entry_id>` 
+2. Go to the `Indicators` and [filter](https://docs.paloaltonetworks.com/cortex/cortex-xsoar/5-5/cortex-xsoar-admin/manage-indicators/understand-indicators/indicators-page.html#idf15421df-a0e0-41fb-b8d4-07d0f610bbec) to find all of the Indicators you extracted from the text file.
```
Comment (ytauman) [inline Packs/EDL/Integrations/EDL/README.md:6]: go to the indicators **page**...
also - how would the user need to filter in order to see all of the relevant indicators?

File: Packs/EDL/Integrations/EDL/README.md [hunk 7]
```diff
@@ -1,13 +1,22 @@
...
+3. If needed, batch select the indicators and add a tag to indicators you which to host as a specific EDL. Use this tag in the `Indicator Query` integration parameter when configuring the integration.
```
Comment (ytauman) [inline Packs/EDL/Integrations/EDL/README.md:7]: and add a tag **to the ones you wish to** host as...

File: Packs/EDL/Integrations/EDL/README.md [hunk 8]
```diff
@@ -1,13 +1,22 @@
...
+1. Create IOCs in Cortex XSOAR from the EDLs data, which was previously hosted on an external web server. To do so automatically, extract your EDL as a text file from the web server, upload it as a file to the war-room and use the `ExtractIndicatorsFromTextFile` automation. e.g, `!ExtractIndicatorsFromTextFile entryID=<entry_id>` 
+2. Go to the `Indicators` and [filter](https://docs.paloaltonetworks.com/cortex/cortex-xsoar/5-5/cortex-xsoar-admin/manage-indicators/understand-indicators/indicators-page.html#idf15421df-a0e0-41fb-b8d4-07d0f610bbec) to find all of the Indicators you extracted from the text file.
+3. If needed, batch select the indicators and add a tag to indicators you which to host as a specific EDL. Use this tag in the `Indicator Query` integration parameter when configuring the integration. 
+4. Maintain your indicators by understanding the (expiration)[https://docs.paloaltonetworks.com/cortex/cortex-xsoar/6-0/cortex-xsoar-admin/manage-indicators/understand-indicators/indicator-expiration.html] policy and the [exclusion](https://docs.paloaltonetworks.com/cortex/cortex-xsoar/6-0/cortex-xsoar-admin/manage-indicators/understand-indicators/exclusion-list.html) list.
```
Comment (ytauman) [inline Packs/EDL/Integrations/EDL/README.md:8]: not sure about this comment here - i think it can be removed

File: Packs/EDL/Integrations/EDL/README.md [hunk 9]
```diff
@@ -1,13 +1,22 @@
...
+5. You can always edit a single or bulk of indicators from the Indicators screen.
```
Comment (ytauman) [inline Packs/EDL/Integrations/EDL/README.md:9]: this one too - it's kind of implied in point 3

File: Packs/EDL/Integrations/EDL/README.md [hunk 10]
```diff
@@ -1,13 +1,22 @@
...
+6. Follow the rest of this guide to make sure that the PAN-OS device has connection to the EDL service.
+7. Upon a configuration of 100s of FWs and a Panorama device, we recommend creating an EDL object in the Panorama device that will be populated from the `PAN-OS EDL Service`. the EDL object should be pushed to its respective FWs.
```
Comment (ytauman) [inline Packs/EDL/Integrations/EDL/README.md:11]: capital T in the second sentence.
+ small rephrasing to the first one:
`Upon a configuration of 100 FWs or more, we recommend using your Panorama device and creating an EDL object there that will be populated from the...`

File: Packs/EDL/Integrations/EDL/README.md (hunk 6 above)
Comment (bakatzir) [reply Packs/EDL/Integrations/EDL/README.md:6]: `First Seen` or any other time filter.

File: Packs/PaloAltoNetworks_PAN_OS_EDL_Management/Integrations/PaloAltoNetworks_PAN_OS_EDL_Management/PaloAltoNetworks_PAN_OS_EDL_Management_description.md (hunk 3 above)
Comment (bakatzir) [reply Packs/PaloAltoNetworks_PAN_OS_EDL_Management/Integrations/PaloAltoNetworks_PAN_OS_EDL_Management/PaloAltoNetworks_PAN_OS_EDL_Management_description.md:9]: usually it is `/var/www/html`, pending the web server.

File: Packs/PaloAltoNetworks_PAN_OS_EDL_Management/Integrations/PaloAltoNetworks_PAN_OS_EDL_Management/PaloAltoNetworks_PAN_OS_EDL_Management_description.md (hunk 1 above)
Comment (bakatzir) [reply Packs/PaloAltoNetworks_PAN_OS_EDL_Management/Integrations/PaloAltoNetworks_PAN_OS_EDL_Management/PaloAltoNetworks_PAN_OS_EDL_Management_description.md:8]: it is the private one. removed ssh.

File: Packs/EDL/Integrations/EDL/README.md (hunk 8 above)
Comment (bakatzir) [reply Packs/EDL/Integrations/EDL/README.md:8]: let's wait for dean/alex. I think it might be tricky as IPS on the EDL live forever, ours live 14 days by default.

Comment (ytauman) [reply Packs/EDL/Integrations/EDL/README.md:8]: when you say 'live' - you mean the indicator, not the EDL.
i went over the pack and it doesn't seem like we even mention expiration anywhere (we don't provide a default query, for example).
the EDL can be configured to whatever query - meaning the user doesn't have to take expiration into consideration.

File: Packs/EDL/Integrations/EDL/README.md (hunk 6 above)
Comment (ytauman) [reply Packs/EDL/Integrations/EDL/README.md]: yeah - i tried this out and saw that there is no indication as to how this indicator was created in the system..
it's actually not that critical, considering it's not the filter the user should be using to maintain the EDL - it's just a way of dumping existing knowledge into XSOAR.

in this case, maybe it's better to rephrase to 'Go to the Indicators page and make sure the indicators were created successfully' - making it a generic saying, without getting into the filtering?

File: Packs/PaloAltoNetworks_PAN_OS_EDL_Management/Integrations/PaloAltoNetworks_PAN_OS_EDL_Management/PaloAltoNetworks_PAN_OS_EDL_Management_description.md (hunk 3 above)
Comment (ytauman) [reply Packs/PaloAltoNetworks_PAN_OS_EDL_Management/Integrations/PaloAltoNetworks_PAN_OS_EDL_Management/PaloAltoNetworks_PAN_OS_EDL_Management_description.md:9]: the point was that i don't understand why this step is needed 😅

Comment (bakatzir) [reply Packs/PaloAltoNetworks_PAN_OS_EDL_Management/Integrations/PaloAltoNetworks_PAN_OS_EDL_Management/PaloAltoNetworks_PAN_OS_EDL_Management_description.md:9]: because when you update the file on the web server, you need to know its path, and the path the pan-os device looks at.

File: Packs/EDL/Integrations/EDL/README.md (hunk 8 above)
Comment (bakatzir) [reply Packs/EDL/Integrations/EDL/README.md:8]: You convinced me. I'm thinking that when migrating from a constant hard coded list to q query of a dynamic DB, we need to point them to more KB so they will feel in control. probably does not belong here.

**Review by DeanArbel — APPROVED** [review]
Good catches in the EDL readme.
Shouldn't EDL be receiving a version bump as well?

**Comment by ShirleyDenkberg** [comment]
@DeanArbel @yuvalbenshalom @ytauman @glicht Doc review completed.
